apps/web/src/state/farmsV4/state/utils.ts

```diff
@@ -12,8 +13,18 @@ export const parseFarmPools = (
   options: { isFarming?: boolean } = {},
 ): PoolInfo[] => {
   return data.map((pool) => {
+    const localFarm = UNIVERSAL_FARMS.find(
```

Comment:
the complexity is almost O(N^2), and you know the data and UNIVERSAL_FARMS are both large object, i think there may cause performance issues.